src/Renci.SshNet/Sftp/SftpFileStream.cs

```diff
@@ -838,29 +818,31 @@ public override long Seek(long offset, SeekOrigin origin)
                     // Abandon the read buffer.
                     _bufferPosition = 0;
                     _bufferLen = 0;
+                }
 
-                    // Seek to the new position.
-                    switch (origin)
-                    {
-                        case SeekOrigin.Begin:
-                            newPosn = offset;
-                            break;
-                        case SeekOrigin.Current:
-                            newPosn = _position + offset;
-                            break;
-                        case SeekOrigin.End:
-                            var attributes = _session.RequestFStat(_handle, false);
-                            newPosn = attributes.Size - offset;
-                            break;
-                    }
-
-                    if (newPosn < 0)
-                    {
-                        throw new EndOfStreamException();
-                    }
+                // Seek to the new position.
+                switch (origin)
+                {
+                    case SeekOrigin.Begin:
+                        newPosn = offset;
+                        break;
+                    case SeekOrigin.Current:
+                        newPosn = _position + offset;
+                        break;
+                    case SeekOrigin.End:
+                        var attributes = _session.RequestFStat(_handle, false);
+                        newPosn = attributes.Size + offset;
+                        break;
+                    default:
+                        throw new ArgumentException("Invalid seek origin.", "origin");
+                }
 
-                    _position = newPosn;
+                if (newPosn < 0)
+                {
+                    throw new EndOfStreamException();
```

Comment:
nit: .NET reference source show use of [IOException](https://referencesource.microsoft.com/mscorlib/system/io/ioexception.cs.html#db370b64091ac4df)
at https://referencesource.microsoft.com/#mscorlib/system/io/memorystream.cs,1a4dcb744a23ba6f

```csharp
if (offset < 0 || tempPosition < _origin)
                    throw new IOException(Environment.GetResourceString("IO.IO_SeekBeforeBegin"));
```

Should SSH.NET use the same exception? EndOfStreamException for seeking before start feels off.

Reply:
I agree, though this may be out of the scope of this PR as I would need to modify every other seek test as well.